lib/fog/libvirt/requests/compute/list_domains.rb

```diff
@@ -39,13 +39,9 @@ def domain_display xml
         end
 
         def domain_volumes xml
-          vols_by_file = xml_elements(xml, "domain/devices/disk/source", "file")
-          vols_by_name = xml_elements(xml, "domain/devices/disk/source", "name")
-          vols = []
-          vols_by_file.zip(vols_by_name).each do |by_file,by_name|
-            vols.push(by_file.nil? ? by_name : by_file)
-          end
-          vols
+          xml_elements(xml, "domain/devices/disk/source").map{|element|
+            element["file"] || element["dev"] || element["name"]
+          }
```

Comment:
Sorry for not catching this previously, but it's common to use `do/end` when you have multiple lines.

I don't have a test env at hand, but in the util function they're retrieved via symbols. Should that also be done here as well?

```suggestion
          xml_elements(xml, "domain/devices/disk/source").map do |element|
            element[:file] || element[:dev] || element[:name]
          end
```

Reply:
> Sorry for not catching this previously, but it's common to use `do/end` when you have multiple lines.

No problem to change this, but then we should change it in ``domain_interfaces`` as well for consistency.

> I don't have a test env at hand, but in the util function they're retrieved via symbols. Should that also be done here as well?

It works with symbols, so no problem changing it, probably also better for consistency. I just used strings because that's used in the Nokogiri rdocs.